File: src/Application Specific/Classes/Entity Collections/Spares/ProductSupplier/ProductSupplierValidator.vb
Namespace Entity

Namespace ProductSuppliers

Public Class ProductSupplierValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oProductSupplier As ProductSupplier)

        'make sure that contact object is valid
        If oProductSupplier.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oProductSupplier.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

                'If oProductSupplier.ProductCode.Trim.Length = 0 Then
                '    Me.AddCriticalMessage("*Product Code Missing")
                'End If

                If oProductSupplier.SupplierID <= 0 Then
                    Me.AddCriticalMessage("*Supplier Missing")
                End If

                If oProductSupplier.QuantityInPack < 1 Then
                    Me.AddCriticalMessage("*Invalid Quantity In Pack")
                End If

                If oProductSupplier.Price <= 0 Then
                    Me.AddCriticalMessage("*Invalid Price")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Spares/ProductSupplier/ProductSupplierValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ProductSuppliers
    {
        // make sure that contact object is valid
        public class ProductSupplierValidator : BaseValidator
        {
            public void Validate(ProductSupplier oProductSupplier)
            {

                // make sure that contact object is valid
                if (oProductSupplier.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oProductSupplier.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                // If oProductSupplier.ProductCode.Trim.Length = 0 Then
                // Me.AddCriticalMessage("*Product Code Missing")
                // End If

                if (oProductSupplier.SupplierID <= 0)
                {
                    AddCriticalMessage("*Supplier Missing");
                }

                if (oProductSupplier.QuantityInPack < 1)
                {
                    AddCriticalMessage("*Invalid Quantity In Pack");
                }

                if (oProductSupplier.Price <= 0)
                {
                    AddCriticalMessage("*Invalid Price");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}